src/libasr/pass/pass_utils.h

```diff
@@ -120,12 +141,73 @@ namespace LCompilers {
                     ASR::is_a<ASR::SymbolicExpression_t>(*ASRUtils::expr_type(var)));
         }
 
-        static inline bool is_symbolic_list_type(ASR::expr_t* var) {
-            if (ASR::is_a<ASR::List_t>(*ASRUtils::expr_type(var))) {
-                ASR::List_t *list = ASR::down_cast<ASR::List_t>(ASRUtils::expr_type(var));
-                return (list->m_type->type == ASR::ttypeType::SymbolicExpression);
```

Comment:
This function is needed as per https://github.com/lcompilers/lpython/commit/b122222fb4c5051e7574575ba56ee16196feee90